lib/board.ex

```diff
@@ -1,57 +1,47 @@
 defmodule Board do
 
-  @markers ["X","O"]
-
-  def create(board_size) do
-    {:x, [["1","2","3"],["4","5","6"],["7","8","9"]], :first_turn}
+  def create do
+    [["1","2","3"],["4","5","6"],["7","8","9"]]
   end
 
-  def place_marker(selected_tile, game_status) do
-    marker = elem(game_status, 0)
-    elem(game_status, 1)
-    |> check_tile_not_taken(selected_tile, marker)
-  end
+  def place_marker(selected_tile, board, player_symbol) do
+    marker_string = Markers.get_player_marker(player_symbol)
+    index_selection = zero_index_selection(selected_tile)
+    row_size = length(board)
 
-  def check_tile_not_taken(board, selected_tile, marker) do
-     selection_value = get_value_of_selected_tile(zero_index_selection(selected_tile), board)
-     Enum.member?(@markers,selection_value)
-     |> create_next_game_status(board, marker, selected_tile)
+    do_replace_marker_in_board(board, index_selection, marker_string, row_size)
   end
 
-  def replace_marker_in_board(board,selected_tile,marker) do
-   marker_string = get_marker_symbol(marker)
-   index_selection = zero_index_selection(selected_tile)
-   row_size = length(board)
-   List.flatten(board)
-   |> List.replace_at(index_selection, marker_string)
-   |> divide_board_into_rows(row_size)
-  end
-
-  def divide_board_into_rows(board, row_size) do
-    Enum.chunk(board, row_size)
-  end
+  def check_tile_not_taken(board, selected_tile) do
+    selection_value = get_value_of_selected_tile(zero_index_selection(selected_tile), board)
 
-  def create_next_game_status(is_duplicate_selection, board, marker, selected_tile) when is_duplicate_selection === false  do
-    new_board = replace_marker_in_board(board, selected_tile, marker)
-    turn_type = WinOrTieChecker.check_if_win_or_tie(new_board)
-    {marker,new_board,turn_type}
-  end
+   if Enum.member?(Markers.get_all_markers,selection_value) do
+     :tile_already_selected
+   else
+     :continue
+   end
 
-  def create_next_game_status(is_duplicate_selection, board, marker, _) when is_duplicate_selection === true do
-    {marker, board,:tile_already_selected}
   end
 
-  def get_marker_symbol(marker) do
-    available_markers = [{:x, "X"}, {:o, "O"}]
-    available_markers[marker]
+  def size(board) do  
+    board
+    |> List.flatten()
+    |> length
   end
 
-  def zero_index_selection(selected_tile) do
-    selected_tile - 1
+  defp do_replace_marker_in_board(board, index_selection, marker_string, row_size) do
+    List.flatten(board)
+    |> List.replace_at(index_selection, marker_string)
+    |> divide_board_into_rows(row_size)
```

Comment:
You could calculate the `row_size` dynamically within `divide_board_into_rows` by doing "_row_size_ = sqrt(board.size) which could be a nice little firend to `Board.size(board)`

This would reduce the number of arguments passed around.